### PR TITLE
M1 build script changes

### PR DESCRIPTION
**Pull request by seibert**
Similar to https://github.com/numba/llvmlite/pull/813

### CODE REVIEW

File: buildscripts/condarecipe.local/meta.yaml [hunk 1]
```diff
@@ -29,7 +29,7 @@ requirements:
     - {{ compiler('cxx') }}    # [not (armv6l or armv7l or aarch64)]
     # both of these are needed on osx, llvm for the headers, Intel for the lib
     - llvm-openmp              # [osx]
-    - intel-openmp             # [osx]
+    - intel-openmp             # [osx and x86_64]
```
Comment (stuartarchibald) [inline buildscripts/condarecipe.local/meta.yaml:32]: xref: https://github.com/numba/numba/issues/7468 there's a reasonably high chance that this dep can just be deleted?

**Comment by stuartarchibald** [comment]
https://github.com/numba/numba/pull/7786 attempts to remove the `intel-openmp` dependency.

**Comment by seibert** [comment]
OK, great.  I'll let that one finish review and merge before updating this PR.

**Comment by seibert** [comment]
I should also note that a few Numba tests fail on my M1, but I don't think those are build related.  (A few look like BLAS issues on ARM, so look forward to that...)

**Comment by esc** [comment]
This branch is now conflicting against `master`.

**Comment by esc** [comment]
I think this pr can be abandoned. It turns out that we can build fine when all the underlying changes to llvmdev and llvmlite have been applied. However, trying to compile and run and test Numba locally on an M1 machine yields multiple errors, which I will open a separate ticket for.

**Comment by stuartarchibald** [comment]
I think https://github.com/numba/numba/pull/7786 renders this patch unneeded as the `intel-openmp` is no longer a dependency.

**Pull request closed without merging**